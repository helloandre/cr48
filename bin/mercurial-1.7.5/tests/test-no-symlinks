#!/bin/sh

"$TESTDIR/hghave" no-symlink || exit 80

# The following script was used to create the bundle:
#
# hg init symlinks
# cd symlinks
# echo a > a
# mkdir d
# echo b > d/b
# ln -s a a.lnk
# ln -s d/b d/b.lnk
# hg ci -Am t
# hg bundle --base null ../test-no-symlinks.hg

# Extract a symlink on a platform not supporting them
echo % unbundle
hg init t
cd t
hg pull -q "$TESTDIR/test-no-symlinks.hg"
hg update

cat a.lnk && echo
cat d/b.lnk && echo

# Copy a symlink and move another
echo % move and copy
hg copy a.lnk d/a2.lnk
hg mv d/b.lnk b2.lnk
hg ci -Am copy
cat d/a2.lnk && echo
cat b2.lnk && echo

# Bundle and extract again
echo % bundle
hg bundle --base null ../symlinks.hg
cd ..

hg init t2
cd t2
hg pull ../symlinks.hg
hg update

cat a.lnk && echo
cat d/a2.lnk && echo
cat b2.lnk && echo
