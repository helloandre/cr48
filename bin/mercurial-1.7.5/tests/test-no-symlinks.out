% unbundle
4 files updated, 0 files merged, 0 files removed, 0 files unresolved
a
d/b
% move and copy
a
d/b
% bundle
2 changesets found
pulling from ../symlinks.hg
requesting all changes
adding changesets
adding manifests
adding file changes
added 2 changesets with 6 changes to 6 files
(run 'hg update' to get a working copy)
5 files updated, 0 files merged, 0 files removed, 0 files unresolved
a
a
d/b
